File: resilient/resilient-annotation-processor/src/test/java/io/koraframework/resilient/annotation/processor/aop/testdata/RateLimitTarget.java
package io.koraframework.resilient.annotation.processor.aop.testdata;

import io.koraframework.common.Component;
import io.koraframework.common.annotation.Root;
import io.koraframework.resilient.ratelimiter.annotation.RateLimit;

@Component
@Root
public class RateLimitTarget {

    @RateLimit("custom1")
    public String getValueSync() {
        return "OK";
    }

    @RateLimit("custom1")
    public void getValueSyncVoid() {
        // no-op
    }
}

File: resilient/resilient-symbol-processor/src/test/kotlin/io/koraframework/resilient/symbol/processor/aop/testdata/RateLimitTarget.kt
package io.koraframework.resilient.symbol.processor.aop.testdata

import io.koraframework.common.Component
import io.koraframework.common.annotation.Root
import io.koraframework.resilient.ratelimiter.annotation.RateLimit

@Component
@Root
open class RateLimitTarget {

    @RateLimit("custom1")
    open fun getValueSync(): String {
        return "OK"
    }

    @RateLimit("custom1")
    open fun getValueSyncVoid() {
        // no-op
    }
}
